craft_application/services/provider.py

```diff
@@ -116,6 +117,13 @@ def instance(
         base = self.get_base(base_name, instance_name=instance_name, **kwargs)
         provider = self.get_provider(name=self.__provider_name)
 
+        if base.alias == DEVEL_BASE:
+            emit.message(
+                "The development build-base should only be used for testing purposes, "
+                "as its contents are bound to change with the opening of new Ubuntu "
+                "releases, suddenly and without warning."
+            )
```

Comment:
The distinction between `emit.message` and `emit.progress` is still unclear to me, but from [the docs](https://canonical-craft-cli.readthedocs-hosted.com/en/latest/explanations/#regular-messages) this seems like it maybe should be:

```suggestion
            emit.progress(
                "The development build-base should only be used for testing purposes, "
                "as its contents are bound to change with the opening of new Ubuntu "
                "releases, suddenly and without warning.",
                permanent=True,
            )
```

Reply:
I'm not sure either.  [This](https://canonical-craft-cli.readthedocs-hosted.com/en/latest/craft_cli.messages/#craft_cli.messages.Emitter.message) describes it similarly, but extends it for "important messages".  I think that warnings fit the "messages" definition slightly more than the "progress" definition since a warning doesn't indicate the progress of anything.

Either way, the UX will be the same so I don't mind which is used.

PS - I recall that `progress(..., permanent=True)` was getting used everywhere because `message()` was broken due to [this](https://github.com/canonical/craft-cli/issues/202) for quite a while.

Reply:
I've asked upstream to clarify. And by upstream I mean us 🙃 

https://github.com/canonical/craft-cli/issues/248